src/overlays/actors/ovl_En_Dinofos/z_en_dinofos.c

```diff
@@ -1459,30 +1490,31 @@ static s8 sLimbToBodyParts[DINOLFOS_LIMB_MAX] = {
 
 void EnDinofos_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfx) {
     EnDinofos* this = THIS;
-    Vec3f sp80;
-    Vec3f sp74;
-    Vec3f sp68;
-    Vec3f sp5C;
-    s32 sp58;
-    Vec3f sp4C;
-    MtxF* sp48;
-
-    Collider_UpdateSpheres(limbIndex, &this->colliderJntSph);
+    Vec3f knifeTipQuadOldPos;
+    Vec3f knifeBaseQuadOldPos;
+    Vec3f knifeTipQuadPos;
+    Vec3f knifeBaseQuadPos;
+    s32 dimAlphaStep;
+    Vec3f fireVelocity;
+    MtxF* matrix;
+
+    Collider_UpdateSpheres(limbIndex, &this->bodyAndFireCollider);
     if (sLimbToBodyParts[limbIndex] != BODYPART_NONE) {
         Matrix_MultZero(&this->bodyPartsPos[sLimbToBodyParts[limbIndex]]);
     }
 
-    if ((limbIndex == DINOLFOS_LIMB_RIGHT_HAND) && (this->unk_292 != this->unk_290) &&
-        ((this->actionFunc == func_8089C56C) || (this->actionFunc == func_8089C2A8))) {
-        Math_Vec3f_Copy(&sp74, &this->colliderQuad.dim.quad[0]);
-        Math_Vec3f_Copy(&sp80, &this->colliderQuad.dim.quad[1]);
-        Matrix_MultVec3f(&D_8089E38C, &sp68);
-        Matrix_MultVec3f(&D_8089E398, &sp5C);
-        Collider_SetQuadVertices(&this->colliderQuad, &sp5C, &sp68, &sp74, &sp80);
-        if (this->colliderQuad.base.atFlags & AT_ON) {
-            EffectBlure_AddVertex(Effect_GetByIndex(this->unk_2A0), &sp68, &sp5C);
+    if ((limbIndex == DINOLFOS_LIMB_RIGHT_HAND) && (this->timer2 != this->timer1) &&
```

Comment:
```suggestion
    if ((limbIndex == DINOLFOS_LIMB_RIGHT_HAND) && (this->timer2 != this->attackTimer) &&
```

Below, you set `timer2` to `attackTimer`:
```c
this->timer2 = this->attackTimer;
```

So you should do the same thing up here too.

I did some research into what the whole point of this check is, and I think I have this one figured out. This whole `timer2` stuff is a bit of a hacky way to ensure that this code only gets run once per Update step. It's possible for a Draw function to be called multiple times for any given Update call (for example, by repeatedly pausing quickly so the game can't advance to the next frame), so if you put gameplay logic in the Draw function like this one does, you have to be careful. This is the cause of a lot of glitches like Epona Push, Sword Hitbox Displacement, etc. I've seen developers do a few things to work around this, but never anything like this. The way it works is that `attackTimer` is updated in an Update step, then `timer2` is updated in the Draw step. If they're matching, that means there hasn't been a new Update step since the last time we drew something, so we shouldn't do anything here.

I have no idea what you would even call `timer2` in this scenario, though. But maybe that gets the creative juices flowing.